### PR TITLE
Publish to PYPI using Github CI 

### PR DESCRIPTION
**Pull request by kmilo9999**
Added workflow to publish to pip

### CODE REVIEW

File: .github/workflows/publish_pypi.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v2
+      
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
```
Comment (hollandjg) [inline .github/workflows/publish_pypi.yml:18]: Are we using python 3.11 in the rest of the project? Perhaps we should use the same here.

File: .github/workflows/publish_pypi.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    - name: Build package
+      run: |
+        python -m build
+
+    - name: Publish to PyPI
+      env:
+        TWINE_USERNAME: "__token__"
+        TWINE_PASSWORD: ${{ secrets.PYPI_TOKEN }}
+      run: |
+        python -m twine upload --skip-existing --repository pypi dist/*
```
Comment (hollandjg) [inline .github/workflows/publish_pypi.yml:34]: If you replace this bit with trusted publishers on PyPI, you can avoid having to deal with tokens https://docs.pypi.org/trusted-publishers/using-a-publisher/ 

There's some more info here: https://intersect-training.org/packaging/09-publishing-citations/index.html

Comment (hollandjg) [inline .github/workflows/publish_pypi.yml:34]: Maybe set this up initially so it publishes to test-pypi for testing?

**Review by hollandjg — CHANGES_REQUESTED** [review]
Great! I've a couple of suggestions.

File: .github/workflows/publish_pypi.yml (hunk 1 above)
Comment (kmilo9999) [reply .github/workflows/publish_pypi.yml]: Changed to python 3.11

File: .github/workflows/publish_pypi.yml (hunk 2 above)
Comment (kmilo9999) [reply .github/workflows/publish_pypi.yml]: added trusable publishers action

Comment (kmilo9999) [reply .github/workflows/publish_pypi.yml]: changed to pypi-test

**Review by hollandjg — APPROVED** [review]
Awesome! 🚀

**Pull request merged**